kornia/utils/integral.py

```diff
@@ -0,0 +1,52 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+
+
+def integral(input: torch.Tensor) -> torch.Tensor:
+    r"""Calculates integral of the input tensor.
+
+    Args:
+        image: the input tensor with shape :math:`(B,C,H,W)` with shape :math:`(B,C,H,W)`.
+
+    Returns:
+        Integral tensor for the input tensor
+
+    Examples:
+        >>> input = torch.randn(2,2,5,5)
+        >>> output = integral(input)
+        >>> output.shape
+        torch.Size([2, 2, 5, 5])
+    """
+    if not isinstance(input, torch.Tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+    S = torch.cumsum(input, dim=-1)
```

Comment:
avoid import torch
```python
S: Tensor = input
S = S.cumsum(-1)
S = S.cumsum(-2)
```
or we could even loop if we want to support 3d images ... maybe for a futur PR

Reply:
If torch import is to be avoided, how to process tensors?

Reply:
the input is torch, you just use the built-in ops, in this case cumsum.

ideally in the midterm future using the same syntax i will work also with numpy
https://numpy.org/doc/stable/reference/generated/numpy.cumsum.html

i say midterm because we need to make compatible all the checks and (not here) special functions like convs